compiler/src/dmd/mars.d

```diff
@@ -1937,10 +1948,41 @@ bool createModules(ref Strings files, ref Strings libmodules, ref Param params,
         modules.push(m);
         if (firstmodule)
         {
-            global.params.objfiles.push(m.objfile.toChars());
+            params.objfiles.push(m.objfile.toChars());
             firstmodule = false;
         }
     }
+
+    // Special module representing `stdin`
+    if (params.readStdin)
+    {
+        Module m;
+        if (createModule("__stdin.d", libmodules, params, target, eSink, m))
```

Comment:
From the source on stdin only, you can still have
```
import __stdin;
```
As a way for the stdin module to import itself.

The only way to guard against this is to put something like a space in the identifier name.

Or we take the `getpid()` code below, so the module name becomes pseudo random - which helps here in reducing the number of "fixups" in post module creation.

Reply:
Oh yes, that's right, there's at least one fail test that matches `__stdin.d(#): some error` :-)